src/lib/options/validate-options.ts

```diff
@@ -0,0 +1,173 @@
+import type { Options } from './index';
+
+const boolOpts = [
+  'makeCumulative',
+  'loop',
+  'showIcons',
+  'showGroups',
+  'mouseControls',
+  'keyboardControls',
+  'autorun',
+  'injectStyles',
+  'highlightBars',
+  'selectBars',
+  'fixedScale',
+] as const;
+const numberOpts = [
+  'labelsWidth',
+  'tickDuration',
+  'topN',
+  'minHeight',
+  'minWidth',
+  'marginTop',
+  'marginRight',
+  'marginBottom',
+  'marginLeft',
+] as const;
+const strOpts = ['theme', 'startDate', 'endDate'] as const;
+const strOrNumberOpts = ['colorSeed', 'inputHeight', 'inputWidth', 'height', 'width'] as const;
+const strOrFuncOpts = ['title', 'subTitle', 'dateCounter', 'caption'] as const;
+
+const validDataShapes = ['long', 'wide'];
+const validDataTypes = ['json', 'csv', 'tsv', 'xml'];
+const validLabelsPositions = ['inside', 'outside'];
+const validControlButtons = ['all', 'play', 'none'];
+const validOverlays = ['all', 'play', 'repeat', 'none'];
+const validFillDateGapsIntervals = [null, 'year', 'month', 'day'];
+const validFillDateGapsValues = ['last', 'interpolate'];
+
+export function validateOptions(options: Partial<Options>): Partial<Options> {
+  const newOptions: Partial<Options> = {};
+
+  // Validate boolean options
+  boolOpts.forEach((opt) => {
+    if (is(options[opt], 'boolean')) {
+      newOptions[opt] = options[opt];
+    } else if (is(options[opt], 'string')) {
+      // Ignore typescript until option types is updated.
+      // @ts-expect-error
+      if (options[opt] === 'true' || options[opt] === 'false') {
+        // @ts-expect-error
+        newOptions[opt] = options[opt] === 'true';
+      }
+    }
+  });
```

Comment:
The build is failing because of this. May be change it to:

```suggestion
  // Validate boolean options
  boolOpts.forEach((opt) => {
    if (is(options[opt], 'boolean')) {
      newOptions[opt] = options[opt];
    } else if (is(options[opt], 'string')) {
      const booleanOption = options[opt] as unknown;
      if (booleanOption === 'true' || booleanOption === 'false') {
        newOptions[opt] = booleanOption === 'true';
      }
    }
  });
```